File: Solutions/318. Maximum Product of Word Lengths/318. Maximum Product of Word Lengths.go
func maxProduct(words []string) int {
    n:=len(words)
    value:=make([]int,n)
    for i:=0;i<n;i+=1{
        tmp:=words[i]
        for j:=0;j<len(tmp);j+=1{
            value[i]|=1<<(tmp[j]-'a')
        }
    }
    res:=0
    for i:=0;i<n;i+=1{
        for j:=i+1;j<n;j+=1{
            if (value[i]&value[j])==0 && (len(words[i])*len(words[j])>res){
                res=len(words[i])*len(words[j])
            }
        }
    }
    return res
}

File: Solutions/318. Maximum Product of Word Lengths/318. Maximum Product of Word Lengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words) {
        int n=words.size();
        vector<int> value(n);
        for(int i=0;i<n;++i){
            auto tmp=words[i];
            value[i]=0;
            for(int j=0;j<tmp.size();++j){
                value[i]|=1<<(tmp[j]-'a');
            }
        }
        int res=0;
        for(int i=0;i<n;++i){
            for(int j=i+1;j<n;++j){
                if((value[i]&value[j])==0 and (words[i].size()*words[j].size()>res)){
                    res=words[i].size()*words[j].size();
                }
            }
        }
        return res;
    }
};